custom/shared/components/HairCheck/HairCheck.js

```diff
@@ -127,6 +131,39 @@ export const HairCheck = () => {
     }
   }, [camError]);
 
+  const showWaitingMessage = useMemo(() => {
+    return (
+      <div className="waiting">
+        <Loader />
```

Comment:
Just a suggestion - I think we can remove the loader when the owner denies the request?

Reply:
It seems the view changes to "you have left the call" when it's denied anyway, so I don't think this actually shows when you get denied 🤔

Reply:
probably okay to leave for now imo, could always be a clean up task later to change the logic here

Reply:
I also think this is okay to leave for now, since the scope of this PR was to focus on Memoization, and I think there's a larger question here about how we want to change the denied experience.